runtime/java.go

```diff
@@ -52,14 +63,25 @@ func (conf javaRuntime) GetBootCmd(cmdConfs map[string]*CmdConfig, env map[strin
 	if conf.Base == "" { // Allow user to use e.g. "openjdk7:java" package instead default one.
 		conf.Base = "openjdk8-zulu-compact1:java"
 	}
-	conf.setDefaultEnv(map[string]string{
-		"XMS":       conf.Xms,
-		"XMX":       conf.Xmx,
-		"CLASSPATH": strings.Join(conf.Classpath, ":"),
-		"JVM_ARGS":  conf.concatJvmArgs(),
-		"MAIN":      conf.Main,
-		"ARGS":      strings.Join(conf.Args, " "),
-	})
+
+	// Only set java-specific environment variables when base is openjdk-like.
+	if isCompatibleBase(conf.Base, javaPackages) {
+		if len(conf.Classpath) == 0 {
+			conf.Classpath = append(conf.Classpath, "/")
+		}
+		if strings.HasSuffix(conf.Main, ".jar") && !util.StringInSlice("-jar", conf.JvmArgs) {
```

Comment:
Please add a test for this path as well.

Reply:
Done. It hurts me that currently we're unable to provide more than one jvm_arg, but hopefully this will be settled soon.